In [1]:
import numpy as np
import statsmodels.api as sm

# 공부시간
X = np.array([1,2,3,4,5])

# 시험점수
y = np.array([50,55,65,70,80])

# 절편항 추가
X2 = sm.add_constant(X)

# 회귀모델
model = sm.OLS(y, X2)

# 학습
result = model.fit()

# 결과
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.987
Model:                            OLS   Adj. R-squared:                  0.982
Method:                 Least Squares   F-statistic:                     225.0
Date:                Mon, 20 Jul 2026   Prob (F-statistic):           0.000643
Time:                        11:40:15   Log-Likelihood:                -8.1084
No. Observations:                   5   AIC:                             20.22
Df Residuals:                       3   BIC:                             19.44
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         41.5000      1.658     25.025      0.0

c:\Python312\Lib\site-packages\statsmodels\stats\stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "


<img src='lec_image/summary.png'>

| 항목                | 의미          |   |               |
| ----------------- | ----------- | - | ------------- |
| coef              | 회귀계수        |   |               |
| std err           | 표준오차        |   |               |
| t                 | t통계량        |   |               |
| P>                | t           |   | 각 변수의 p-value |
| R²                | 결정계수        |   |               |
| Adj R²            | 수정된 결정계수    |   |               |
| F-statistic       | 모델 전체 검정    |   |               |
| Prob(F-statistic) | 모델의 p-value |   |               |


```text
R² = 0.987 → 공부시간이 시험점수 변동의 98.7%를 설명.
회귀계수 = 7.5 → 공부시간이 1시간 늘어나면 시험점수는 평균 7.5점 증가.
p-value = 0.001 → 공부시간은 시험점수에 통계적으로 유의한 영향을 미치는 변수.
95% 신뢰구간 = [5.909, 9.091] → 실제 기울기는 약 5.9~9.1점 사이로 추정됨.
Prob(F-statistic) = 0.000643 → 회귀모델 전체도 통계적으로 유의하다

In [3]:
## p-value 만 보기
print(result.pvalues)

[0.00013991 0.00064312]



| 변수    | 회귀계수(coef) | P-value |        95% 신뢰구간 |
| ----- | ---------: | ------: | --------------: |
| const |     41.500 |   0.000 | 36.223 ~ 46.777 |
| x1    |      7.500 |   0.001 |   5.909 ~ 9.091 |

**x1(공부시간)**

---

#  x1의 p-value = 0.001

귀무가설(H₀)

> **공부시간은 시험점수에 영향을 주지 않는다.**
>

$ \beta_1 = 0 $

대립가설(H₁)

> **공부시간은 시험점수에 영향을 준다.**
>
$ \beta_1 \neq 0 $


---

### 결과

```
P>|t| = 0.001
```


> **공부시간이 실제로 아무 영향도 없는데도 현재와 같은 데이터를 우연히 얻을 확률이 0.1% 정도**라는 뜻

보통 기준은

* p < 0.05 → 유의함
* p < 0.01 → 매우 유의함
* p < 0.001 → 매우 강한 증거

여기서는

```
0.001 < 0.05
```


**귀무가설을 기각**

> **공부시간은 시험점수를 설명하는 유의한 변수.**


In [4]:
## 회귀계수
print(result.params)

[41.5  7.5]


In [5]:
# 신뢰구간
print(result.conf_int())

[[36.22250984 46.77749016]
 [ 5.90877685  9.09122315]]


| 변수        | 회귀계수 |    95% 신뢰구간 |
| --------- | ---: | ----------: |
| 절편(const) | 41.0 | 36.22 ~ 46.77 |
| 공부시간(x1)  |  7.5 |   5.908 ~ 9.091 |


In [6]:
pred = result.predict([1, 6])   # [절편, 공부시간]
print(pred)

[86.5]
